cmd/tsbs_load_clickhouse/process.go

```diff
@@ -0,0 +1,370 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+	_ "github.com/kshvakov/clickhouse"
+	"github.com/timescale/tsbs/load"
+)
+
+type syncCSI struct {
+	// Map hostname to tags.id for this host
+	m     map[string]int64
+	mutex *sync.RWMutex
+}
+
+func newSyncCSI() *syncCSI {
+	return &syncCSI{
+		m:     make(map[string]int64),
+		mutex: &sync.RWMutex{},
+	}
+}
+
+// globalSyncCSI is used when data is not hashed by some function to a worker consistently so
+// therefore all workers need to know about the same map from hostname -> tags_id
+var globalSyncCSI = newSyncCSI()
+
+// subsystemTagsToJSON converts equations as
+// a=b
+// c=d
+// into JSON STRING '{"a": "b", "c": "d"}'
+func subsystemTagsToJSON(tags []string) string {
+	json := "{"
+	for i, t := range tags {
+		args := strings.Split(t, "=")
+		if i > 0 {
+			json += ","
+		}
+		json += fmt.Sprintf("\"%s\": \"%s\"", args[0], args[1])
+	}
+	json += "}"
+	return json
+}
+
+// insertTags fills tags table with values
+func insertTags(db *sqlx.DB, startId int, tagRows [][]string, returnResults bool) map[string]int64 {
+	// Map hostname to tags_id
+	ret := make(map[string]int64)
+
+	// reflect tags table structure which is
+	// CREATE TABLE tags(
+	//	 created_at timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+	//	 id,
+	//   %s
+	// ) engine=MergeTree(created_at, (%s), 8192)
+
+	// build insert-multiple-rows INSERT statement like:
+	// INSERT INTO tags (
+	//   ... list of column names ...
+	// ) VALUES
+	// ( ... row 1 values ... ),
+	// ( ... row 2 values ... ),
+	// ...
+	// ( ... row N values ... ),
+
+	// Columns. Ex.:
+	// hostname,region,datacenter,rack,os,arch,team,service,service_version,service_environment
+	cols := tableCols["tags"]
+	// Add id column to prepared statement
+	sql := fmt.Sprintf(`
+		INSERT INTO tags(
+			id,%s
+		) VALUES (
+			?%s
+		)
+		`,
+		strings.Join(cols, ","),
+		strings.Repeat(",?", len(cols)))
+	if debug > 0 {
+		fmt.Printf(sql)
+	}
+
+	// In a single transaction insert tags row-by-row
+	// ClickHouse driver accumulates all rows inside a transaction into one batch
+	tx, err := db.Begin()
+	if err != nil {
+		panic(err)
+	}
+	stmt, err := tx.Prepare(sql)
+	if err != nil {
+		panic(err)
+	}
+	defer stmt.Close()
+
+	id := startId
+	for _, tagRow := range tagRows {
+		// id of the bew tag
+		id++
+
+		// unfortunately, it is not possible to pass a slice into variadic function of type interface
+		// more details on the item:
+		// https://blog.learngoprogramming.com/golang-variadic-funcs-how-to-patterns-369408f19085
+		// Passing a slice to variadic param with an empty-interface
+		// +1 is here for id
+		var ihellos []interface{} = make([]interface{}, len(tagRow)+1)
```

Comment:
Even though ihellos is just a placeholder to wrangle a slice into a variadic function call, I think we should change the variable names from `hello` and `ihellos` here to something that makes a bit more sense in this context ;-)

Reply:
Yes, this piece of code is not nice. However, it is not easy to name this work-around with something reasonable and descriptive at the same time. Description work-around naming is a challenge by itself. Hope `variadicFriendlyRow` would make a reasonable compromise
Updated

Reply:
```
ihellos := []interface{id}
ihellos = append(ihellos, tagRow...)
```
doesn't work, so no major changes in this part